O/ONNXRuntime/ONNXRuntimeProvidersCUDA/build_tarballs.jl

```diff
@@ -0,0 +1,87 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "ONNXRuntimeProvidersCUDA"
+version = v"1.10.0"
+
+# Cf. https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#requirements
+cuda_version = v"11.4"
+cudnn_version = v"8.2.4"
+
+cuda_compat = "$(cuda_version.major).$(cuda_version.minor)"
+cudnn_compat = string(cudnn_version.major)
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/microsoft/onnxruntime.git", "0d9030e79888d1d5828730b254fedc53c7b640c1"),
+    ArchiveSource("https://github.com/microsoft/onnxruntime/releases/download/v$version/onnxruntime-win-x64-gpu-$version.zip", "0da11b8d953fad4ec75f87bb894f72dea511a3940cff2f4dad37451586d1ebbc"; unpack_target="onnxruntime-x86_64-w64-mingw32")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cuda_version=`echo $bb_full_target | sed -E -e 's/.*cuda\+([0-9]+\.[0-9]+).*/\1/'`
+
+cd $WORKSPACE/srcdir
+
+if [[ $target == x86_64-w64-mingw32* ]]; then
+    chmod 755 onnxruntime-$target/onnxruntime-*/lib/*
+    mkdir -p $includedir $libdir
+    cp -av onnxruntime-$target/onnxruntime-*/include/* $includedir
+    cp -av onnxruntime-$target/onnxruntime-*/lib/*onnxruntime_providers_cuda* $libdir
+    install_license onnxruntime-$target/onnxruntime-*/LICENSE
+else
+    export PATH=$prefix/cuda/bin:$PATH
+    
+    cd onnxruntime
+    git submodule update --init --recursive
+    mkdir -p build
+    cd build
+    cmake $WORKSPACE/srcdir/onnxruntime/cmake \
+        -DCMAKE_INSTALL_PREFIX=$prefix \
+        -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_CUDA_RUNTIME_LIBRARY=Shared \
+        -DCUDA_RUNTIME_LIBRARY=Shared \
+        -DCUDA_USE_STATIC_CUDA_RUNTIME=OFF \
+        -DCMAKE_TRY_COMPILE_TARGET_TYPE=STATIC_LIBRARY \
+        -DCMAKE_CUDA_COMPILER_ID_RUN=1 \
```

Comment:
Just checking if I can get past the CMake CUDA compiler check - the compiler check depends on linking with `cudart_static`, but even though there should be vars for forcing/choosing to link with the dynamic CUDA runtime, it seems like the only thing working is skipping the CMake CUDA compiler check with `CMAKE_CUDA_COMPILER_ID_RUN`.

References:
* https://gitlab.kitware.com/cmake/cmake/-/issues/17559
* https://gitlab.kitware.com/cmake/cmake/-/merge_requests/4178

Reply:
Dropped attempt to build with CUDA from source for now.